numba/parfors/parfor_lowering.py

```diff
@@ -1621,16 +1621,17 @@ def load_potential_tuple_var(x):
                     array_strides.append(strides[j])
                 builder.store(builder.bitcast(ary.data, byte_ptr_t), dst)
         else:
+            aty_is_bool = aty is types.boolean or isinstance(aty, types.BooleanLiteral)
```

Comment:
Shorter and almost as fast as:
```suggestion
            aty_is_bool = isinstance(aty, types.Boolean)
```

Reply:
@eric-wieser Yes, I thought about this, but that way it would be less restrictive (any other type that user will subclass from types.Boolean will match and that type might actually use different data model) and with original version we restrict to types actually using BooleanModel that defines 1b width.

Reply:
If you're worried about that, then you should worry about literal subclasses too:

```suggestion
            # do not allow any other subclasses
            aty_is_bool = type(aty) in (types.Boolean, types.BooleanLiteral)
```